Изменения №2 от 24.05.2023

ZhosanDenis opened the pull request:
- создан контроллер для работы с личным кабинетом пользователя - AccountController, а также пустой сервис на будущее - AccountService и AccountServiceImpl;
- созданы классы User и NewPassword для работы с личным кабинетом пользователя;
- создан контроллер для работы с объявлениями и комментариями к объявлениям - AdController, а также пустой сервис на будущее - AdService и AdServiceImpl;
- созданы классы Ads, CreateAds, FullAds, ResponseWrapperAds для работы с объявлениями;
- созданы классы Comment, CreateComment, ResponseWrapperComment для работы с комментариями;
- в AuthController добавлена документация swagger (описание эндпоинтов);
- добавлена зависимость для подключения Swagger API.

ZhosanDenis committed:
Изменения №2 от 24.05.2023
- создан контроллер для работы с личным кабинетом пользователя - AccountController, а также пустой сервис на будущее - AccountService и AccountServiceImpl;
- созданы классы User и NewPassword для работы с личным кабинетом пользователя;
- создан контроллер для работы с объявлениями и комментариями к объявлениям - AdController, а также пустой сервис на будущее - AdService и AdServiceImpl;
- созданы классы Ads, CreateAds, FullAds, ResponseWrapperAds для работы с объявлениями;
- созданы классы Comment, CreateComment, ResponseWrapperComment для работы с комментариями;
- в AuthController добавлена документация swagger (описание эндпоинтов);
- добавлена зависимость для подключения Swagger API.

dmitry-bizin commented on pom.xml:
```diff
@@ -52,6 +52,11 @@
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.springdoc</groupId>
```
в целом это не обязательно, ведь документация была дана уже, а вы её просто продублируете снова)

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdController.java:
```diff
@@ -0,0 +1,226 @@
+    )
+    @PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    @ResponseBody
+    public ResponseEntity<Ads> addAdvertising(@ModelAttribute CreateAds properties,
```
не, здесь либо аннотации RequestParam надо использовать для каждой части, либо RequestPart

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdController.java:
```diff
@@ -0,0 +1,226 @@
+            tags = "Объявления"
+    )
+    @PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    @ResponseBody
```
ResponseBody не нужна

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AuthController.java:
```diff
@@ -23,20 +27,35 @@ public class AuthController {
+    )
     @PostMapping("/login")
-    public ResponseEntity<?> login(@RequestBody LoginReq req) {
+    public ResponseEntity<?> login(@RequestBody(required = false) LoginReq req) {
```
а почему required = false?

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AuthController.java:
```diff
@@ -23,20 +27,35 @@ public class AuthController {
     @PostMapping("/register")
-    public ResponseEntity<?> register(@RequestBody RegisterReq req) {
-        Role role = req.getRole() == null ? USER : req.getRole();
+    public ResponseEntity<?> register(@RequestBody(required = false) RegisterReq req) {
```
а почему required = false?

ZhosanDenis committed:
Изменения №1 от 28.05.2023
- AccountController: убраны свойства необязательного наличия тела запроса при обновлении пароля и обновлении информации о пользователе;
- AuthController: убраны свойства необязательного наличия тела запроса при регистрации и входе пользователя;
- изменены аннотации параметров при добавлении объявления, убрано свойство необязательного наличия тела запроса при добавлении комментария к объявлению;
- generated-requests.http: изменен http-запрос для добавления объявления с учетом одновременной передачи JSON и изображения.
